### PR TITLE
Add support for custom lazy load

### PR DESCRIPTION
**Pull request by strausr**

### CODE REVIEW

File: src/cloudinary.js [hunk 1]
```diff
@@ -693,6 +693,32 @@ class Cloudinary {
     return this;
   }
 
+  /**
+   * Sets src to null when not using native lazy load
+   * @param img
+   * @param options
+   */
+  handleLazyLoad(img, options){
```
Comment (asisayag2) [inline src/cloudinary.js]: @strausr , there's a alignment between the docString and the method name. 
If what the docString says this method is doing is right, then let's rename the method. 
`handleLazyLoad` sounds like it's doing the entire logic

File: src/cloudinary.js [hunk 2]
```diff
@@ -693,6 +693,32 @@ class Cloudinary {
...
+    if(options.loading && !this.isNativeLazyLoadSupported() && this.isLazyLoadSupported()) {
+      img.setAttribute('src', null);
+      img.setAttribute('width', img.getAttribute('data-width'));
+    }
+  }
+
+  /**
+   * @returns {boolean}
```
Comment (asisayag2) [inline src/cloudinary.js]: @strausr , please elaborate more in the docString

File: src/cloudinary.js [hunk 3]
```diff
@@ -693,6 +693,32 @@ class Cloudinary {
...
+   */
+  isLazyLoadSupported() {
+    return window && 'IntersectionObserver' in window;
+  }
+
+  /**
+   * @returns {boolean}
```
Comment (asisayag2) [inline src/cloudinary.js]: @strausr , please elaborate more in the docString

File: test/spec/lazy-load-spec.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+describe("Lazy Loaded Image", function() {
+  it("should return empty src on Firefox", function() {
+    cl = cloudinary.Cloudinary.new({cloud_name: 'sdk-test'});
+    var img;
+    let options = {loading: "lazy", width: "300"}
+    img = cl.image("sample", {
+      options
+    });
+    cl.handleLazyLoad(img, options);
```
Comment (asisayag2) [inline test/spec/lazy-load-spec.js]: @strausr , why is that needed to be called?

File: src/cloudinary.js (hunk 1 above)
Comment (strausr) [reply src/cloudinary.js]: This we'll currently be only used on the Angular SDK. When our Lazy load will be implemented on React and Vue it will be used there as well.

File: test/spec/lazy-load-spec.js (hunk 4 above)
Comment (strausr) [reply test/spec/lazy-load-spec.js]: The method is expecting an img so we're creating one using image()
